roles/basic_setup_rhel/tasks/main.yml

```diff
@@ -0,0 +1,94 @@
+---
+- name: "Subscription Manifest"
+  include_role:
+    name: theforeman.foreman.manifest
+
+- name: "Enable RHEL7 repository"
+  include_role:
+    name: theforeman.foreman.repositories
+  vars:
+    foreman_products:
+      - name: Red Hat Enterprise Linux Server
+        repository_sets:
+          - name: Red Hat Enterprise Linux 7 Server (RPMs)
+            basearch: x86_64
+            releasever: 7Server
+  when: foreman_basic_setup_rhel_enable_rhel7
+
+- name: "Enable RHEL8 repositories"
+  include_role:
+    name: theforeman.foreman.repositories
+  vars:
+    foreman_products:
+      - name: Red Hat Enterprise Linux for x86_64
+        repository_sets:
+          - name: Red Hat Enterprise Linux 8 for x86_64 - BaseOS (RPMs)
+            releasever: 8
+          - name: Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)
+            releasever: 8
+  when: foreman_basic_setup_rhel_enable_rhel8
+
+- name: "Sync RHEL7 repository"
+  theforeman.foreman.repository_sync:
+    username: "{{ username }}"
+    password: "{{ password }}"
+    organization: "{{ foreman_organization }}"
+    server_url: "{{ server_url }}"
+    validate_certs: "{{ validate_certs }}"
+    product: Red Hat Enterprise Linux Server
+  async: "{{ ansible_check_mode | ternary(0, 14400) }}"
+  poll: 0
+  register: rhel7_sync
+  when: foreman_basic_setup_rhel_enable_rhel7 and foreman_basic_setup_rhel_sync_rhel7_now
+
+- name: "Sync RHEL8 repositories"
+  theforeman.foreman.repository_sync:
+    username: "{{ username }}"
+    password: "{{ password }}"
+    organization: "{{ foreman_organization }}"
+    server_url: "{{ server_url }}"
+    validate_certs: "{{ validate_certs }}"
+    product: Red Hat Enterprise Linux for x86_64
+  async: "{{ ansible_check_mode | ternary(0, 14400) }}"
```

Comment:
This is a workaround for the checkmode test on CI, however there may be a better workaround such as exempting this role from the test altogether

Reply:
Hmmm, at least it is failing differently now :)